.github/workflows/docker-publish.yml

```diff
@@ -48,16 +50,35 @@ jobs:
       # https://github.com/docker/metadata-action
       - name: Extract Docker metadata
         id: meta
-        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        uses: docker/metadata-action@v3
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
 
       # Build and push Docker image with Buildx (don't push on PR)
       # https://github.com/docker/build-push-action
       - name: Build and push Docker image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        uses: docker/build-push-action@v2
         with:
           context: .
+          file: Dockerfile
           push: ${{ github.event_name != 'pull_request' }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Render Action Metadata
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          ./bin/render_action_metadata "docker://ghcr.io/paper-spa/pages:${{ github.ref_name }}" > action.yml
+          git add action.yml
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git commit -m "Add generated action.yml 🤖"
+          if [[ '${{ github.ref_type }}' == 'tag' ]]; then 
+            echo "Pushing tag ${{ github.ref_name }}"
+            git tag -f ${{ github.ref_name }}
+            git push -f origin ${{ github.ref_name }}
+          else
+            echo "Pushing branch ${{ github.ref_name }}"
+            git push origin ${{ github.ref_name }}
+          fi
```

Comment:
nit: I like how most other run steps have simplified down to running a single script with arguments.  Would be nice if this block could be simplified similarly.

Reply:
Yeah I've thought about putting this into a script that can be referenced here, but wanted to make sure this was even on the right track before I continued to "refactor."

Reply:
Something to revisit, definitely not a blocker.  🚀